tls/s2n_connection.h

```diff
@@ -297,6 +299,9 @@ struct s2n_connection {
 
     /* Cookie extension data */
     struct s2n_stuffer cookie_stuffer;
+
+    /* Key update data */
+    uint8_t key_update_pending;
```

Comment:
Can this be a single bit, like `server_name_used` above?

I see this flag being set, but I don't see it being checked (or reset, which I expected for a pending scenario). Is this being checked somewhere and I just don't see it, or is it not needed?

Reply:
This is a flag that is needed for the send functionality. It is not needed for the receive functionality so I can remove to the next pr. And yes, it can be a single bit.